api/src/main/java/org/openmrs/module/appointments/service/impl/RecurringAppointmentServiceImpl.java

```diff
@@ -125,9 +129,72 @@ private Appointment updateMetadata(Appointment pendingAppointment, Appointment a
         Date endTime = getEndTime(pendingAppointment, appointment);
         pendingAppointment.setStartDateTime(startTime);
         pendingAppointment.setEndDateTime(endTime);
+        pendingAppointment.setService(appointment.getService());
+        pendingAppointment.setServiceType(appointment.getServiceType());
+        pendingAppointment.setLocation(appointment.getLocation());
+        pendingAppointment.setComments(appointment.getComments());
+        if (appointment.getProviders() != null) {
+            mapProvidersForAppointment(pendingAppointment,
+                    getDeepCloneOfProviders(new ArrayList<>(appointment.getProviders())));
+        }
         return pendingAppointment;
     }
 
+    private List<AppointmentProvider> getDeepCloneOfProviders(List<AppointmentProvider> appointmentProviders) {
+        List<AppointmentProvider> appointmentProviderList = appointmentProviders
+                .stream()
+                .map(appointmentProvider -> new AppointmentProvider(appointmentProvider))
```

Comment:
You can use `.map(AppointmentProvider::new)`, makes it little simpler..

Reply:
Modified the same.